tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -61,13 +63,44 @@ var XTablesCmds = sets.New(
 	constants.IP6TABLESSAVE,
 )
 
+// XTablesWriteCmds contains all xtables commands that do write actions (and thus need a lock)
+var XTablesWriteCmds = sets.New(
+	constants.IPTABLES,
+	constants.IP6TABLES,
+	constants.IPTABLESRESTORE,
+	constants.IP6TABLESRESTORE,
+)
+
 // RealDependencies implementation of interface Dependencies, which is used in production
 type RealDependencies struct {
+	IptablesVersion  *utilversion.Version
 	NetworkNamespace string
-	HostNSEnterExec  bool
 	CNIMode          bool
 }
 
+const iptablesVersionPattern = `v([0-9]+(\.[0-9]+)+)`
+
+func DetectIptablesVersion(ver string) (*utilversion.Version, error) {
```

Comment:
This should probably additionally check if we're dealing with `iptables-legacy` or `iptables-nft`.

If the latter, regardless of the version #, there is no `xtables` lock anyway, and we can just categorically skip.

Reply:
Yeah I suppose so. Darn nft making everything more complex